metricbeat/module/elasticsearch/pending_tasks/data_test.go

```diff
@@ -80,24 +66,18 @@ func TestInvalidJsonForRequiredFieldShouldThrowError(t *testing.T) {
 	content, err := ioutil.ReadFile(file)
 	assert.NoError(t, err)
 
-	_, errs := eventsMapping(content)
-
-	errors, ok := errs.(*s.Errors)
-	if ok {
-		assert.True(t, errors.HasRequiredErrors(), "mapping error: %s", errors)
-		assert.EqualError(t, errors, "Required fields are missing: ,source")
-	} else {
-		t.Error(err)
-	}
+	_, err = eventsMapping(content)
+	assert.Error(t, err)
 }
 
 func TestInvalidJsonForBadFormatShouldThrowError(t *testing.T) {
+	t.Skip("the code is not really checking for uknown fields")
+
```

Comment:
This test was passing because `schema.Apply()` returned the empty `schema.Errors`, it was not detecting the unknown field in the (valid) JSON file. This test can be probably removed if the missing field is optional, if is not optional, an explicit check should be added in `eventsMapping` to check that it exists.

Reply:
Assuming this is not optional, explicit check added and skip removed.